ambry-router/src/main/java/com/github/ambry/router/GetBlobOperation.java

```diff
@@ -845,6 +848,14 @@ void checkAndMaybeComplete() {
         chunkCompleted = true;
       }
       if (chunkCompleted) {
+        if (quotaChargeCallback != null) {
```

Comment:
For the actions that are done from within checkAndMaybeComplete in all classes, I would suggest that we call the callback only if the completion status is not already ChunkState.Complete:

```
if (state != ChunkState.Complete && quotaChargeCallback != null) { chargeQuota }
```

This comment applies to all operation classes

Reply:
done.